### ISSUE TITLE
Incrementally compile release notes

### ISSUE THREAD

**Issue by BD103** [bevyengine/bevy-website#969]
Originally from [Discord](https://discord.com/channels/691052431525675048/692572690833473578/1208038415015092225).

It would be amazing if release notes could be incrementally compiled, so there is always a draft page of migration guides and release notes for the next version. This would reduce the amount of effort required for each release, so Alice and Cart can spend more time developing and less drafting. This would work well with our deployment process, which [already automatically runs daily](https://github.com/bevyengine/bevy-website/blob/d17be0dafb1b322a0c2a4ca0029ed65864b2fa66/.github/workflows/deploy.yml#L6-L7).

**Comment by TrialDragon** [comment]
A random thought; I wonder if it's worth adding a `Release Notes` section to PRs on the main repo like the `Migration Guide` section so that folks can document what's relevant while it's fresh in their heads. Then either, a stub is generated like with migration guide or something along those lines, or maybe someone just brings over the relevant release notes as they're deemed relevant.

TL;DR Make release notes workflow similar to current migration guide workflow to minimize work needed.

**Comment by BD103** [comment]
That's an interesting idea.a

For this issue specifically, I was more thinking of building the migration guide from `latest...main` nightly, so you could view all breaking changes as they were made.

**Comment by TrialDragon** [comment]
Apparently even as public draft the articles will show in the RSS feed since they _must_ have a date to be rendered due to the overarching sections `sort_by = "date"`. So that may be an issue.

**Comment by alice-i-cecile** [comment]
This is now effectively a reality, as part of #1163. I'd still like to add further automation, but I'm going to call this "done".

**Comment by BD103** [comment]
I agree, thanks for finding this again!

**Comment by alice-i-cecile** [comment]
I added the `C-Automation` label so I was looking for things to apply it to!
